### PR TITLE
Move DTOs to bee-message

### PR DESCRIPTION
**Pull request by thibault-martinez**
## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

**Review by Thoralf-M — DISMISSED** [review]

**Review by Thoralf-M — DISMISSED** [review]

**Review by samuel-rufi — DISMISSED** [review]

**Review by Thoralf-M — DISMISSED** [review]

File: bee-message/src/dto.rs [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::milestone::MilestoneIndex;
+
+use primitive_types::U256;
+use serde::{Deserialize, Serialize};
+
+/// Describes an U256.
```
Comment (Alex6323) [inline bee-message/src/dto.rs]:
```suggestion
/// Describes a U256.
```

File: bee-message/src/dto.rs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+impl TryFrom<&U256Dto> for U256 {
+    type Error = prefix_hex::Error;
+
+    fn try_from(value: &U256Dto) -> Result<Self, Self::Error> {
+        prefix_hex::decode(&value.0)
+    }
+}
+
+#[allow(clippy::trivially_copy_pass_by_ref)]
+pub fn is_zero(num: &u32) -> bool {
```
Comment (Alex6323) [inline bee-message/src/dto.rs:28]: Why do we want to pass the `u32` by reference?

Comment (thibault-martinez) [reply bee-message/src/dto.rs:28]: It's expected by serde's `skip_serializing_if` signature.

File: bee-message/src/message.rs [hunk 3]
```diff
@@ -220,3 +220,70 @@ fn verify_payload(payload: Option<&Payload>) -> Result<(), Error> {
...
+        #[serde(rename = "protocolVersion")]
+        pub protocol_version: u8,
+        #[serde(rename = "parentMessageIds")]
+        ///
+        pub parents: Vec<String>,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        ///
+        pub payload: Option<PayloadDto>,
+        ///
+        pub nonce: String,
```
Comment (Alex6323) [inline bee-message/src/message.rs:245]: missing docs :trollface:

Comment (thibault-martinez) [reply bee-message/src/message.rs:245]: Yeah not doing it in this PR, simply moving stuff around.

**Review by samuel-rufi — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**